src/Qir/Execution/Tools/Utility/ILogger.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Microsoft.Quantum.Qir.Utility
+{
+    /// <summary>
+    /// Logger for internal traces and errors.
+    /// </summary>
+    /// <remarks>For now, this thinly wraps console logging. In the future, this might be extended to log to files or other systems.</remarks>
+    public interface ILogger
```

Comment:
Yes, this mainly needed for error handling in the service. However, I believe this can be removed too, and for tests in this repo we can use the `FileLogger` class and the controller in the service can implement the custom logger.